File: app/src/main/java/com/example/richtext/bind/RichEditTextBindings.java
package com.example.richtext.bind;

import android.app.Activity;
import android.content.Context;
import android.databinding.BindingAdapter;

import com.example.richtext.R;
import com.shuyu.textutillib.EmojiLayout;
import com.shuyu.textutillib.RichEditText;
import com.shuyu.textutillib.model.TopicModel;
import com.shuyu.textutillib.model.UserModel;

public class RichEditTextBindings {

    @BindingAdapter("atResult")
    public static void setAtResult(RichEditText richEditText, UserModel userModel) {
        if (userModel != null) {
            richEditText.resolveAtResult(userModel);
        }
    }

    @BindingAdapter("atResultByEnter")
    public static void setAtResultByEnterAt(RichEditText richEditText, UserModel userModel) {
        if (userModel != null) {
            richEditText.resolveAtResultByEnterAt(userModel);
        }
    }

    @BindingAdapter("topicResultByEnter")
    public static void setTopicResultByEnter(RichEditText richEditText, TopicModel topicModel) {
        if (topicModel != null) {
            richEditText.resolveTopicResultByEnter(topicModel);
        }
    }

    @BindingAdapter("topicResult")
    public static void setTopicResult(RichEditText richEditText, TopicModel topicModel) {
        if (topicModel != null) {
            richEditText.resolveTopicResult(topicModel);
        }
    }
    @BindingAdapter("emojiRichTextView")
    public static void setEmojiRichTextView(EmojiLayout emojiLayout, Context context) {
        if (emojiLayout != null && context != null && context instanceof Activity) {
            Activity activity = (Activity) context;
            emojiLayout.setEditTextSmile((RichEditText)
                    activity.findViewById(R.id.mvvm_rich_edit_text));
        }
    }


}

File: app-kotlin/src/main/java/com/example/shuyu/bind/RichEditTextBindings.kt
package com.example.shuyu.bind

import android.app.Activity
import android.content.Context
import android.databinding.BindingAdapter
import com.example.shuyu.R

import com.shuyu.textutillib.EmojiLayout
import com.shuyu.textutillib.RichEditText
import com.shuyu.textutillib.model.TopicModel
import com.shuyu.textutillib.model.UserModel


open class RichEditTextBindings{

    companion object {
        @BindingAdapter("atResult")
        fun setAtResult(richEditText: RichEditText, userModel: UserModel?) {
            if (userModel != null) {
                richEditText.resolveAtResult(userModel)
            }
        }

        @BindingAdapter("atResultByEnter")
        fun setAtResultByEnterAt(richEditText: RichEditText, userModel: UserModel?) {
            if (userModel != null) {
                richEditText.resolveAtResultByEnterAt(userModel)
            }
        }

        @BindingAdapter("topicResultByEnter")
        fun setTopicResultByEnter(richEditText: RichEditText, topicModel: TopicModel?) {
            if (topicModel != null) {
                richEditText.resolveTopicResultByEnter(topicModel)
            }
        }

        @BindingAdapter("topicResult")
        fun setTopicResult(richEditText: RichEditText, topicModel: TopicModel?) {
            if (topicModel != null) {
                richEditText.resolveTopicResult(topicModel)
            }
        }

        @BindingAdapter("emojiRichTextView")
        fun setEmojiRichTextView(emojiLayout: EmojiLayout?, context: Context?) {
            if (emojiLayout != null && context != null && context is Activity) {
                val activity = context as Activity?
                emojiLayout.setEditTextSmile(activity!!.findViewById(R.id.mvvm_rich_edit_text) as RichEditText)
            }
        }

    }

}
